src/test/java/server/SocketStub.java

```diff
@@ -0,0 +1,27 @@
+package test.java.server;
+
+import java.net.Socket;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+
+class SocketStub extends Socket {
+    private final OutputStream outputStream;
+    private final InputStream inputStream;
+
+    SocketStub(String inputStub) {
+        this.outputStream = new ByteArrayOutputStream();
+        this.inputStream = new ByteArrayInputStream(inputStub.getBytes());
+    }
+
+    @Override
+    public InputStream getInputStream() {
+        return inputStream;
+    }
+
+    @Override
+    public OutputStream getOutputStream() {
+        return outputStream;
+    }
+}
```

Comment:
I like this. Seems pretty safe since you'd get a compiler error if Socket ever changed its `getInputStream` or `getOutputStream` methods.